src/puppetlabs/puppetdb/cli/benchmark.clj

```diff
@@ -292,36 +295,92 @@
+         (contains? options :nummsgs))
+    (do
+      (utils/println-err
+        "Error: -N/--nummsgs runs immediately and is not compatable with -i/--runinterval")
+      (action-on-error-fn))
+
+    (and (contains? options :archive)
+         (some #{:reports :catalogs :facts :default} (keys options)))
+    (do
+      (utils/println-err
+        "Error: -A/--archive is incompatible with -F/--facts, -C/--catalogs, -R/--reports, -D/--default.")
+      (action-on-error-fn))
+
+    (and (contains? options :default)
+         (some #{:reports :catalogs :facts :archive} (keys options)))
+    (do
+      (utils/println-err
+        "Error: -D/--archive is incompatible with -F/--facts, -C/--catalogs, -R/--reports, -A/--archive.")
+      (action-on-error-fn))
+    :else options))
+
+(defn default-options
+  [options]
+  (-> options
+      (assoc :facts "acceptance/benchmark/facts/")))
+
 (defn- validate-cli!
   [args]
   (try+
-   (ks/cli! args supported-cli-options required-cli-options)
-   (catch map? m
-     (println (:message m))
-     (case (:type m)
-       :puppetlabs.kitchensink.core/cli-error (System/exit 1)
-       :puppetlabs.kitchensink.core/cli-help (System/exit 0)))))
-
-(defn validate-nummsgs [options action-on-error-fn]
-  (when (and (contains? options :runinterval)
-             (contains? options :nummsgs))
-    (utils/println-err "Error: -N/--nummsgs runs immediately and is not compatable with -i/--runinterval")
-    (action-on-error-fn)))
+    (let [options (-> (ks/cli! args supported-cli-options required-cli-options)
+                      first
+                      (validate-options #(System/exit 1)))]
+      (if (:default options)
+        (default-options options)
+        options))
+
+    (catch map? m
+      (println (:message m))
+      (case (:type m)
+        :puppetlabs.kitchensink.core/cli-error (System/exit 1)
+        :puppetlabs.kitchensink.core/cli-help (System/exit 0)))))
+
+(defn process-tar-entry
+  [archive-path tar-reader]
+  (fn [acc entry]
+    (let [path (.getName entry)
+          catalog-pattern "catalogs.*\\.json$"
+          report-pattern "reports.*\\.json$"
+          facts-pattern "facts.*\\.json$"
+          parsed-entry (json/parse-string (archive/read-entry-content tar-reader))]
+      (cond
+        (re-find (re-pattern catalog-pattern) path) (update-in acc [:catalogs] #(cons parsed-entry %))
+        (re-find (re-pattern report-pattern) path) (update-in acc [:reports] #(cons parsed-entry %))
+        (re-find (re-pattern facts-pattern) path) (update-in acc [:facts] #(cons parsed-entry %))
+        :else acc))))
+
+(defn build-tar-map
+  [path]
+  (let [tar-reader (archive/tarball-reader path)
+        entries (archive/all-entries tar-reader)]
+    (reduce (process-tar-entry path tar-reader)
+            {:catalogs [] :reports [] :facts []} entries)))
 
 (defn -main
   [& args]
-  (let [[options _]     (validate-cli! args)
+  (let [options         (validate-cli! args)
```

Comment:
I've cleaned it up a bit with load-data-from-options, should do the trick I think.